docs/development.rst

```diff
@@ -33,11 +33,19 @@ Quickstart
 - Build the low level module by running ``make`` in the ``python`` directory.
 - Run the tests to ensure everything has worked: ``python -m nose -vs``. These should
   all pass.
-
-- Make your changes in a local branch, and open a pull request on GitHub when you
-  are ready. Please make sure that (a) the tests pass before you open the PR; and
-  (b) your code passes PEP8 checks (see :ref:`sec_development_code_style` below
-  for a git commit hook to ensure this happens automatically) before opening the PR.
+- Install the pre-commit checks: ``pre-commit install``
+- Make your changes in a local branch. On each commit a "pre-commit hook" will run
+  checks for code style and common problems (see :ref:`sec_development_code_style`).
+  Sometimes these will report "files were modified by this hook" ``git add``
+  and ``git commit --amend`` will update the commit with the automatically modified
+  version.
+  The modifications made are for consistency, code readability and designed to
+  minimise merge conflicts. They are guaranteed not to modify the functionality of the
+  code. To run the checks without committing use ``pre-commit run``. To bypass
+  the checks (to save or get feedback on work-in-progress) use ``git commit
+  --no-verify``
+- When ready open a pull request on GitHub. Please make sure that the tests pass before
+  you open the PR.
```

Comment:
Don't we want people to be able to ask for help in getting tests to pass as part of the PR?

Reply:
I've added an extra note to that effect.